dask/dataframe/core.py

```diff
@@ -4555,14 +4556,80 @@ def sort_values(
             **kwargs,
         )
 
+    def _check_index(self, other: Any) -> Optional[Any]:
+        """Internal method to check the column passed to set_index.
+
+        Returns None if this column is already the index.
+        """
+
+        # Check other can be translated to column name or column object
+        index = other
+        if not isinstance(index, str):
+            if isinstance(index, Sequence):
```

Comment:
I'm not sure that this should be imported from `typing` if you're using it in an `isinstance()` check, can you comment on that choice so I can learn? Thank you 😄

Reply:
The problem with `collections.abc.Sequence` is that you cannot do `Sequence[str]` in Python 3.8 (from 3.9 onwards they've unified this and the `typing` versions). So there are some here options:
* Using `typing.Sequence` in all cases (it works, but is not very clean).
* Using `typing.Sequence` for annotation, and checking `isinstance` explicitly against `collections.abc.Sequence` (from Python 3.9 onwards it suffices with just `collections.abc.Sequence`).
* Importing `typing.Sequence` and `collections.abc.Sequence` with 2 aliases so as to avoid clashes, but it is confusing anyhow.

Reply:
> The problem with `collections.abc.Sequence` is that you cannot do `Sequence[str]` in Python 3.8

Can you do this with `__future__.annotations`? If not, I think I prefer importing with two separate names, perhaps `import typing.Sequence as SequenceType`. @crusaderky @ian-r-rose How have we handled this in the past?

Reply:
Did the latter. One in `SequenceType` and the other is left as plain `Sequence`.

Reply:
Please don't double-import.

Please use 
```python
from __future__ import annotations
from collections.abc import Sequence
```
*unless* you need to subclass a generic; in which case import from typing
```python
from typing import Sequence

class MySeq(Sequence[int]):
    ....

Reply:
Now removed this. Since I was doing `Sequence[Any]` for `divisions`, I simply left that as pure `Sequence`, so there's no need to import `typing.Sequence` anymore, but just `collections.abc.Sequence`.